### ISSUE TITLE
Metric definition ignored in Sklearn0.23-1 training

### ISSUE THREAD

**Issue by gerold-csendes-epam** [aws/sagemaker-python-sdk#2275]
**Describe the bug**
Metric_definitons parameter of the [EstimatorBase](https://sagemaker.readthedocs.io/en/stable/api/training/estimators.html#sagemaker.estimator.EstimatorBase) class gets ignored when submitting a training job using the SKLearn 0.23-1 framework.

**To reproduce**
Write training file
```python
%%writefile train_iris.py

from sklearn.linear_model import LogisticRegression
from sklearn.datasets import load_iris

if __name__ == "__main__": 

    iris = load_iris()
    X = iris.data[:, :2]  # we only take the first two features.
    Y = iris.target

    # Create an instance of Logistic Regression Classifier and fit the data.
    logreg = LogisticRegression(C=1e5)
    logreg.fit(X, Y)

    print(f"Accuracy={logreg.score(X, Y)}")
```

Submit training job:
```python
from sagemaker.sklearn.estimator import SKLearn
from sagemaker import get_execution_role

role = get_execution_role()
metric_definitions = [{'Name': 'Accuracy',
                       'Regex': 'Accuracy=([0-9\\.]+)'}]

sklearn_estimator = SKLearn(
    entry_point='train_iris.py',
    framework_version='0.23-1', 
    instance_type="ml.m5.xlarge",
    role=role,
    base_job_name=TRAINING_BASE_NAME,
    metric_definitons=metric_definitions)

sklearn_estimator.fit()
```

**Expected behavior**
I would expect to see the MetricDefinitions show up in `sklearn_estimator.latest_training_job.describe()["AlgorithmSpecification"]` but it does not nor in the Training UI on AWS.

**Screenshots or logs**
If applicable, add screenshots or logs to help explain your problem.

**System information**
A description of your system. Please provide:
- **SageMaker Python SDK version**: 2.31.1
- **Framework name (eg. PyTorch) or algorithm (eg. KMeans)**:  SKLearn
- **Framework version**: 0.23-1
- **Python version**: py3
- **CPU or GPU**: CPU
- **Custom Docker image (Y/N)**: N

**Comment by gerold-csendes-epam** [comment]
This is the output of my pip freeze: 
```
aiobotocore @ file:///tmp/build/80754af9/aiobotocore_1615487469391/work
aiohttp @ file:///home/conda/feedstock_root/build_artifacts/aiohttp_1614280892404/work
aioitertools @ file:///home/conda/feedstock_root/build_artifacts/aioitertools_1605140085294/work
async-timeout==3.0.1
attrs @ file:///home/conda/feedstock_root/build_artifacts/attrs_1605083924122/work
backcall @ file:///home/conda/feedstock_root/build_artifacts/backcall_1592338393461/work
backports.functools-lru-cache==1.6.1
bcrypt==3.2.0
boto3 @ file:///home/conda/feedstock_root/build_artifacts/boto3_1616540878956/work
botocore @ file:///home/conda/feedstock_root/build_artifacts/botocore_1616536672126/work
brotlipy==0.7.0
category-encoders @ file:///home/conda/feedstock_root/build_artifacts/category_encoders_1588199510095/work/dist
certifi==2020.12.5
cffi @ file:///home/conda/feedstock_root/build_artifacts/cffi_1606236101993/work
chardet @ file:///home/conda/feedstock_root/build_artifacts/chardet_1610093490430/work
cryptography @ file:///home/conda/feedstock_root/build_artifacts/cryptography_1615405999564/work
decorator==4.4.2
distro==1.5.0
docker==4.4.4
docker-compose==1.28.6
dockerpty==0.4.1
docopt==0.6.2
fsspec @ file:///home/conda/feedstock_root/build_artifacts/fsspec_1614107550004/work
google-pasta==0.2.0
idna==2.10
importlib-metadata @ file:///home/conda/feedstock_root/build_artifacts/importlib-metadata_1615778734765/work
ipykernel @ file:///home/conda/feedstock_root/build_artifacts/ipykernel_1613852938555/work/dist/ipykernel-5.5.0-py3-none-any.whl
ipython @ file:///home/conda/feedstock_root/build_artifacts/ipython_1614393573755/work
ipython-genutils==0.2.0
jedi @ file:///home/conda/feedstock_root/build_artifacts/jedi_1610146791023/work
jmespath @ file:///home/conda/feedstock_root/build_artifacts/jmespath_1589369830981/work
joblib @ file:///home/conda/feedstock_root/build_artifacts/joblib_1612898609989/work
jsonschema==3.2.0
jupyter-client @ file:///home/conda/feedstock_root/build_artifacts/jupyter_client_1615693636836/work
jupyter-core @ file:///home/conda/feedstock_root/build_artifacts/jupyter_core_1612125275706/work
mock @ file:///home/conda/feedstock_root/build_artifacts/mock_1610094566888/work
multidict @ file:///home/conda/feedstock_root/build_artifacts/multidict_1610319019165/work
numexpr @ file:///home/conda/feedstock_root/build_artifacts/numexpr_1614971340455/work
numpy @ file:///home/conda/feedstock_root/build_artifacts/numpy_1612953005674/work
packaging @ file:///home/conda/feedstock_root/build_artifacts/packaging_1612459636436/work
pandas==1.2.3
paramiko==2.7.2
parso @ file:///home/conda/feedstock_root/build_artifacts/parso_1607618318316/work
patsy==0.5.1
pexpect @ file:///home/conda/feedstock_root/build_artifacts/pexpect_1602535608087/work
pickleshare @ file:///home/conda/feedstock_root/build_artifacts/pickleshare_1602536217715/work
prompt-toolkit @ file:///home/conda/feedstock_root/build_artifacts/prompt-toolkit_1616432837031/work
protobuf==3.15.6
protobuf3-to-dict @ file:///home/conda/feedstock_root/build_artifacts/protobuf3-to-dict_1610115095957/work
ptyprocess @ file:///home/conda/feedstock_root/build_artifacts/ptyprocess_1609419310487/work/dist/ptyprocess-0.7.0-py2.py3-none-any.whl
pycparser @ file:///home/conda/feedstock_root/build_artifacts/pycparser_1593275161868/work
Pygments @ file:///home/conda/feedstock_root/build_artifacts/pygments_1615243893546/work
PyNaCl==1.4.0
pyOpenSSL @ file:///home/conda/feedstock_root/build_artifacts/pyopenssl_1608055815057/work
pyparsing==2.4.7
pyrsistent==0.17.3
PySocks @ file:///home/conda/feedstock_root/build_artifacts/pysocks_1610291447907/work
python-dateutil==2.8.1
python-dotenv==0.15.0
pytz @ file:///home/conda/feedstock_root/build_artifacts/pytz_1612179539967/work
PyYAML==5.4.1
pyzmq @ file:///home/conda/feedstock_root/build_artifacts/pyzmq_1614611703677/work
requests==2.25.1
s3fs @ file:///home/conda/feedstock_root/build_artifacts/s3fs_1608058532532/work
s3transfer @ file:///home/conda/feedstock_root/build_artifacts/s3transfer_1616166519874/work
sagemaker==2.31.1
scikit-learn @ file:///home/conda/feedstock_root/build_artifacts/scikit-learn_1605985735074/work
scipy @ file:///home/conda/feedstock_root/build_artifacts/scipy_1615955928702/work
six @ file:///home/conda/feedstock_root/build_artifacts/six_1590081179328/work
smdebug-rulesconfig @ file:///home/conda/feedstock_root/build_artifacts/smdebug-rulesconfig_1608558075041/work
statsmodels @ file:///home/conda/feedstock_root/build_artifacts/statsmodels_1612273599609/work
tables @ file:///home/conda/feedstock_root/build_artifacts/pytables_1610156075390/work
texttable==1.6.3
threadpoolctl @ file:///tmp/tmp79xdzxkt/threadpoolctl-2.1.0-py3-none-any.whl
tornado @ file:///home/conda/feedstock_root/build_artifacts/tornado_1610094706440/work
traitlets @ file:///home/conda/feedstock_root/build_artifacts/traitlets_1602771532708/work
typing-extensions @ file:///home/conda/feedstock_root/build_artifacts/typing_extensions_1602702424206/work
urllib3 @ file:///home/conda/feedstock_root/build_artifacts/urllib3_1615828766818/work
wcwidth @ file:///home/conda/feedstock_root/build_artifacts/wcwidth_1600965781394/work
websocket-client==0.58.0
wrapt @ file:///home/conda/feedstock_root/build_artifacts/wrapt_1610094884173/work
yarl @ file:///home/conda/feedstock_root/build_artifacts/yarl_1610354134018/work
zipp @ file:///home/conda/feedstock_root/build_artifacts/zipp_1614945704755/work
```

**Comment by NeilJonkers** [comment]
Seems you have a spelling error when defining: metric_definitions. Please try the estimator corrected code below:

```
sklearn_estimator = SKLearn(
    entry_point='train_iris.py',
    framework_version='0.23-1', 
    instance_type="ml.m5.xlarge",
    role=role,
    base_job_name=TRAINING_BASE_NAME,
    metric_definitions=metric_definitions)
```

**Comment by gerold-csendes-epam** [comment]
@NeilJonkers thanks for checking the code. I fixed the typo and now the metric definition shows up in the training job ui:
![scikit-estimator-metric](https://user-images.githubusercontent.com/74710495/114375471-f565a600-9b84-11eb-9509-8e9f8e7e0f56.PNG)
I am still wondering on two things:

1. Why this didn't produce any error/warning.
2. Why the metric value shows up nowhere in the UI. I would expect to find that in a similar fashion as for the hyperparameter tuning jobs as shown below. Am I missing something?

![hyperparam-objective](https://user-images.githubusercontent.com/74710495/114376076-98b6bb00-9b85-11eb-93cd-5d8d435288c9.PNG)

**Comment by NeilJonkers** [comment]
On Question 1: 'Why this didn't produce any error/warning.'

The EstimatorBase Class [here](https://sagemaker.readthedocs.io/en/stable/api/training/estimators.html#sagemaker.estimator.EstimatorBase) take 
Python keyword argument arguments **kwargs: Notice [here](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/sklearn/estimator.py#L46)

Example in python:

```
 def testfunc(a='test', b='news', **kwargs):
     print(a,b)

 testfunc(a='foo',b='bar',somevar='foobar')
```
Here the function testfunc does not utilise **somevar**.
Python will not throw error or exception in this scenario.

Question 2: 'Why the metric value shows up nowhere in the UI.'
In the Monitor section for the training job you will notice your metric definition will be shown in a graph.
This Metric data is saved to Cloudwatch.

I hope this help.

**Comment by gerold-csendes-epam** [comment]
thanks @NeilJonkers this does help. Closing this issue.
